language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/codeaction/AddModuleToBallerinaTomlCodeActionTest.java

```diff
@@ -0,0 +1,117 @@
+import org.ballerinalang.langserver.commons.workspace.WorkspaceManager;
+import org.ballerinalang.langserver.contexts.LanguageServerContextImpl;
+import org.ballerinalang.langserver.extensions.ballerina.connector.CentralPackageListResult;
+import org.ballerinalang.langserver.util.FileUtils;
+import org.ballerinalang.langserver.util.TestUtil;
+import org.ballerinalang.langserver.workspace.BallerinaWorkspaceManager;
+import org.eclipse.lsp4j.jsonrpc.Endpoint;
+import org.mockito.Mockito;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Test class to test module addition to Ballerina.toml.
+ *
+ * @since 2201.8.0
+ */
+public class AddModuleToBallerinaTomlCodeActionTest extends AbstractCodeActionTest {
+    
+    @Override
+    protected void setupLanguageServer(TestUtil.LanguageServerBuilder builder) {
+        builder.withInitOption(InitializationOptions.KEY_POSITIONAL_RENAME_SUPPORT, true);
+    }
+
+    @BeforeClass
+    public void setup() {
+        super.setup();
+
+        LanguageServerContext context = new LanguageServerContextImpl();
+        BallerinaLanguageServer languageServer = new BallerinaLanguageServer();
+        Endpoint endpoint = TestUtil.initializeLanguageSever(languageServer);
+        try {
+            Map<String, String> localProjects = Map.of("pkg1", "main.bal", "pkg2", "main.bal");
+            List<LSPackageLoader.ModuleInfo> localPackages = getPackages(localProjects,
+                    languageServer.getWorkspaceManager(), context).stream().map(LSPackageLoader.ModuleInfo::new)
+                    .collect(Collectors.toList());
+            Mockito.when(getLSPackageLoader().getLocalRepoPackages(Mockito.any())).thenReturn(localPackages);
+        } catch (Exception e) {
+            //ignore
+        } finally {
+            TestUtil.shutdownLanguageServer(endpoint);
+        }
+    }
+
+    @Test(dataProvider = "codeaction-data-provider")
+    @Override
+    public void test(String config) throws IOException, WorkspaceDocumentException {
+        super.test(config);
+    }
+
+    @DataProvider(name = "codeaction-data-provider")
+    @Override
+    public Object[][] dataProvider() {
+        return new Object[][]{
+                {"add_module1.json"},
+                {"add_module2.json"},
+                {"add_module3.json"},
+                {"add_module4.json"},
+                {"add_module5.json"},
+        };
+    }
+
+    @Override
+    public String getResourceDir() {
+        return "add-module-ballerina-toml";
+    }
+
+    @Override
+    public boolean loadMockedPackages() {
+        return true;
+    }
+
+    protected static List<Package> getPackages(Map<String, String> projects,
```

Comment:
We can remove this as per above comment

Reply:
https://github.com/ballerina-platform/ballerina-lang/pull/41146#discussion_r1282904808

Reply:
Let's add the Override annotation

Reply:
We can't override this method, right?

Reply:
We have the same method in AbstractLSTest class. So we have to override that or otherwise we can use the method in the super class and remove this method

Reply:
Renamed the method